### ISSUE TITLE
Exceptions starting notebooks x 163 REGRESSION?

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#2334]
When running the configuration & start notebooks exceptions similar to the following are seen

```
lization
Thu Jan 02 09:51:18 GMT 2020 cocoMDS5 Information OMRS-AUDIT-0106 Refreshing registration with open metadata repository cohort cocoCohort using metadata collection id 4d82b40a-c516-405c-b8ab-4cf0da9487e0 at the request of server cocoMDS3
Thu Jan 02 09:51:18 GMT 2020 cocoMDS3 Security OPEN-METADATA-SECURITY-0015 User cocoMDS3npa is not authorized to change open metadata type EmbeddedConnection (eb6dfdd2-8c6f-4f0d-a17d-f6ce4799f64f) on server cocoMDS3
Thu Jan 02 09:51:18 GMT 2020 cocoMDS3 Exception OMRS-AUDIT-9019 The type definition event processor for the cocoCohort cohort inbound service caught an unexpected exception org.odpi.openmetadata.repositoryservices.ffdc.exception.UserNotAuthorizedException with message OMAG-PLATFORM-SECURITY-403-011 User cocoMDS3npa is not authorized to change open metadata type EmbeddedConnection (eb6dfdd2-8c6f-4f0d-a17d-f6ce4799f64f) on server cocoMDS3
Thu Jan 02 09:51:18 GMT 2020 cocoMDS3 Exception OMRS-AUDIT-9019 a19b1d5a-6536-44fd-a3f1-a16f569e1ebd org.odpi.openmetadata.repositoryservices.ffdc.exception.UserNotAuthorizedException returned message of OMAG-PLATFORM-SECURITY-403-011 User cocoMDS3npa is not authorized to change open metadata type EmbeddedConnection (eb6dfdd2-8c6f-4f0d-a17d-f6ce4799f64f) on server cocoMDS3 and stacktrace of
UserNotAuthorizedException{userId='cocoMDS3npa', reportedHTTPCode=403, reportingClassName='org.odpi.openmetadata.metadatasecurity.samples.CocoPharmaServerSecurityConnector', reportingActionDescription='validateUserForTypeUpdate', errorMessage='OMAG-PLATFORM-SECURITY-403-011 User cocoMDS3npa is not authorized to change open metadata type EmbeddedConnection (eb6dfdd2-8c6f-4f0d-a17d-f6ce4799f64f) on server cocoMDS3', reportedSystemAction='The system is unable to process a request from the user because they do not have access to the necessary services and/or resources.', reportedUserAction='The request fails with a UserNotAuthorizedException exception.', reportedCaughtException=UserNotAuthorizedException{userId='cocoMDS3npa', reportedHTTPCode=403, reportingClassName='org.odpi.openmetadata.metadatasecurity.samples.CocoPharmaServerSecurityConnector', reportingActionDescription='validateUserForTypeUpdate', errorMessage='OMAG-PLATFORM-SECURITY-403-011 User cocoMDS3npa is not authorized to change open metadata type EmbeddedConnection (eb6dfdd2-8c6f-4f0d-a17d-f6ce4799f64f) on server cocoMDS3', reportedSystemAction='The system is unable to process a request from the user because they do not have access to the necessary services and/or resources.', reportedUserAction='The request fails with a UserNotAuthorizedException exception.', reportedCaughtException=null, relatedProperties=null}, relatedProperties=null}
```
We have 64 from cocoMDS2npa, 40 from cocoMDS3npa, 32 from cocoMDS5npa & 8 from cocoMDS6npa

All are UserNotAuthorized exceptions. As such this may be expected - the only concern is the additional stack trace information in the audit log is likely confusing for the reader. But additionally this looks like it's more of a versioning/type issue? since it relates to a type update and could be a regression

**Comment by planetf1** [comment]
full log at https://gist.github.com/bd6ec676719de74a9eba04e429310076

**Comment by planetf1** [comment]
There are also 2 null pointer exceptions
```
Thu Jan 02 09:51:34 GMT 2020 cocoMDS4 Information OMRS-AUDIT-0050 The Open Metadata Repository Services (OMRS) is about to process open metadata archive Open Metadata Types
Thu Jan 02 09:51:34 GMT 2020 cocoMDS4 Exception OMRS-AUDIT-9019 The type definition event processor for the ODPi Egeria (1.3) service caught an unexpected exception java.lang.NullPointerException with message null
Thu Jan 02 09:51:34 GMT 2020 cocoMDS4 Exception OMRS-AUDIT-9019 1e243409-55ee-470f-a09c-c27a0e75c9f3 java.lang.NullPointerException returned message of null and stacktrace of
java.lang.NullPointerException
        at org.odpi.openmetadata.repositoryservices.localrepository.repositorycontentmanager.OMRSRepositoryContentManager.processUpdatedTypeDefEvent(OMRSRepositoryContentManager.java:2646)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processTypeDefStore(OMRSArchiveManager.java:345)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataArchive(OMRSArchiveManager.java:226)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataTypes(OMRSArchiveManager.java:139)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.setLocalRepository(OMRSArchiveManager.java:102)
        at org.odpi.openmetadata.repositoryservices.admin.OMRSOperationalServices.initialize(OMRSOperationalServices.java:419)
        at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithSuppliedConfig(OMAGServerOperationalServices.java:328)
        at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithStoredConfig(OMAGServerOperationalServices.java:149)
        at org.odpi.openmetadata.adminservices.spring.OperationalServicesResource.activateWithStoredConfig(OperationalServicesResource.java:41)
```
```
Thu Jan 02 09:51:34 GMT 2020 cocoMDS4 Exception OMRS-AUDIT-9019 The type definition event processor for the ODPi Egeria (1.3) service caught an unexpected exception java.lang.NullPointerException with message null
Thu Jan 02 09:51:34 GMT 2020 cocoMDS4 Exception OMRS-AUDIT-9019 1ee072d2-e3ba-4cc0-b327-1a2fc43bd5fa java.lang.NullPointerException returned message of null and stacktrace of
java.lang.NullPointerException
        at org.odpi.openmetadata.repositoryservices.localrepository.repositorycontentmanager.OMRSRepositoryContentManager.processUpdatedTypeDefEvent(OMRSRepositoryContentManager.java:2646)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processTypeDefStore(OMRSArchiveManager.java:345)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataArchive(OMRSArchiveManager.java:226)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.processOpenMetadataTypes(OMRSArchiveManager.java:139)
        at org.odpi.openmetadata.repositoryservices.archivemanager.OMRSArchiveManager.setLocalRepository(OMRSArchiveManager.java:102)
        at org.odpi.openmetadata.repositoryservices.admin.OMRSOperationalServices.initialize(OMRSOperationalServices.java:419)
        at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithSuppliedConfig(OMAGServerOperationalServices.java:328)
        at org.odpi.openmetadata.adminservices.OMAGServerOperationalServices.activateWithStoredConfig(OMAGServerOperationalServices.java:149)
        at org.odpi.openmetadata.adminservices.spring.OperationalServicesResource.activateWithStoredConfig(OperationalServicesResource.java:41)
```

**Comment by planetf1** [comment]
May be related to #2326

**Comment by mandy-chessell** [comment]
It looks like thes are 3 different issues.  The NotAuth is due to the OpenMetadataSecurity sample not allowing the servers to change types.  I have changed the updateTypeDef authorization check to be the same as the newTypeDef authorization check so that the metadata architects and the metadata servers can change types.

We also discussed whether it is appropriate for this type of error to push the stack trace to the console and agreed it should not.   So I have also changed the logging of architected checked exceptions - which typically include a reasonable message - to log as an error rather than exception.

The NPE is a bug in the event processor code that occurs when there is no local repository.  I have changed the update typedef logic to handle no local repository connector.  We only see this in cocoMDS4 because all of the other servers have a local repository.

Finally we talked about these error coming out in the console
<img width="1106" alt="Screenshot 2020-01-02 at 10 56 43" src="https://user-images.githubusercontent.com/12354195/71667251-30f7aa80-2d5c-11ea-8943-7b364c3cd2fd.png">

This is also a bug.  The code to cache a type is checking the new type does not have a different version number to any existing version stored - this is clearly not true in the case of a patch, so the code logs that the versions are different.  There is no need to do this validation check as the type have been validated already (if Egeria has got this far with an invalid type def then we have bigger problems :)
